lib/Lift/CodeGenerator.cpp

```diff
@@ -1351,6 +1239,7 @@ void CodeGenerator::translate(Optional<uint64_t> RawVirtualAddress) {
     NewSegment.IsReadable = S.IsReadable;
     NewSegment.IsWriteable = S.IsWriteable;
     NewSegment.IsExecutable = S.IsExecutable;
+    NewSegment.CustomName = S.generateName();
```

Comment:
We don't want this, `CustomName` is optional, aimed at user's customization and it should not be used to generate variable names (linker injection stuff).

Reply:
yeah, but what name do we give to the segment then when we invoke clang?

Reply:
Currently we generated a deterministic name `segement-$(StartAddress.toString())-$(EndAddress.toString())` and we use it as the name of the global variable and of the section. `CustomName` is for end user pretty printing only, in the IR we want deterministic names.